_posts/en/newsletters/2021-01-13-newsletter.md

```diff
@@ -0,0 +1,213 @@
+
+[Add unit testing of node eviction logic][review club
+#20477] is a PR ([#20477][Bitcoin Core #20477]) by
+[practicalswift][practicalswift] to improve test coverage of Bitcoin Core's peer
+eviction logic when a node's inbound connection slots are full. Extreme care
+must be taken to avoid exposing the node to attacker-triggered network
+partitioning with this logic, so this code is critical.
+
+Most of the discussion focused on understanding Bitcoin Core's peer eviction
+logic.
+
+{% include functions/details-list.md
+  q0="Inbound or outbound peer selection: which one is our primary defense
+      against eclipse attacks?"
+  a0="Outbound peer selection, as outbound peers are far less under attacker
+      control than inbound connections. Inbound peer eviction is a second-order
+      protection---and not all nodes permit incoming connections."
+  a0link="https://bitcoincore.reviews/20477#l-77"
+
+  q1="Why does Bitcoin Core evict inbound connections?"
+  a1="To make inbound slots available for honest peers in the network so that
+      new nodes can establish good outbound connections to them. Otherwise,
+      dishonest nodes could more easily attack new nodes by being available for
+      their outbound connections and by statically occupying inbound slots."
+  a1link="https://bitcoincore.reviews/20477#l-66"
+
+  q2="When it needs to free up a slot, how does Bitcoin Core decide which
+      inbound peer to evict?"
+  a2="Up to 28 peers are protected from eviction based on criteria that are
+      difficult to forge, such as low latency, network group, providing novel
+      valid transactions and blocks, and a few others. The longest-connected
+      remaining half are protected, including some onion peers. Of those that
+      remain, the youngest member of the network group with the most
+      connections is selected for disconnection. An attacker would have to be
+      simultaneously better than honest peers at all of these characteristics
+      to partition a node."
+  a2link="https://bitcoincore.reviews/20477#l-83"
+
+  q3="Starting from what number of inbound peers does eviction begin to occur?"
+  a3="Starting from 21 inbound peers, as at least 20 are protected by the
+      current logic: 4 by network group, 4 that sent us novel blocks passing
+      initial validity checks, 4 that sent us novel transactions accepted into
+      our mempool, and 8 having the lowest minimum ping time (best latency)."
+  a3link="https://bitcoincore.reviews/20477#l-218"
+%}
+
+## Releases and release candidates
+
+*New releases and release candidates for popular Bitcoin infrastructure
+projects.  Please consider upgrading to new releases or helping to test
+release candidates.*
+
+FIXME:harding to update Tuesday
+
+- [Bitcoin Core 0.21.0rc5][Bitcoin Core 0.21.0] is a release candidate
+  for the next major version of this full node implementation and its
+  associated wallet and other software.
```

Comment:
Would we consider adding the [0.21 testing guide](https://github.com/bitcoin-core/bitcoin-devwiki/wiki/0.21-Release-Candidate-Testing-Guide)?